Lib/test/test_io.py

```diff
@@ -502,6 +499,8 @@ class UnseekableWriter(self.MockUnseekableIO):
             (self.BytesIO, "rws"), (self.StringIO, "rws"),
         )
         for [test, abilities] in tests:
+            if test == pipe_writer and not threading_helper.can_start_thread:
+                continue
```

Comment:
Should this use `skipTest` rather than `continue`?

Reply:
If we wanted to do that we'd have to split the `with` statement below like:
```
            with self.subTest(test):
                if cond:
                   skipTest()
                with test() as obj:
                     ...
```
and then we'd have to indent the whole body. I think it's better to use `continue` than to have to indent a bunch of stuff another level.

Reply:
Agree that it requires some code churn (and some deeper-than-ideal nesting). 

My motivation here is to make sure that we have visibility on the skip - with a continue, there's no tracking that the test wasn't executed. If the code ever skips for the wrong reason, or you need to audit to see what *isn't* being tested, there's no visibility on that.

How about a third option: Pull out the current test logic as a `_subtest()` utility method that takes `test` as an argument. That way, the indent level for the body of the test stays the same; the only code churn is to move the for loop to *after* the subtest utility method. However, that for loop becomes a much clearer "run this subtest for each configuration" loop, with clear "run or skip" logic for each case.

Reply:
I think this makes sense to me I'll try it out.

Reply:
Though if we're skipping because of threading I guess the only question is whether the feature in question could be tested without threading. Other than that there's nothing here to fix.

Reply:
Agreed that's another way to fix this... but I don't see an obvious way to implement the pipe writer test without threading (at least, not without leaving the test prone to blocking). I'm open to suggestions, though.

Reply:
@freakboy3742 how's this look?

Reply:
With "hide whitespace changes" the diff is quite small.

Reply:
Looks great - with one tweak. Since we're outside a loop here, `return` seems more appropriate than `continue`. 
```suggestion
                return
```

Reply:
TIL it's possible to have a `continue` in an inner function...